app/src/features/surveys/components/EditSamplingMethod.tsx

```diff
@@ -0,0 +1,54 @@
+import EditDialog from 'components/dialog/EditDialog';
+import yup from 'utils/YupSchema';
+import MethodForm, { IEditSurveySampleMethodData, ISurveySampleMethodData } from './MethodForm';
+
+interface IEditSamplingMethodProps {
+  open: boolean;
+  initialData?: IEditSurveySampleMethodData;
+  onSubmit: (data: ISurveySampleMethodData, index?: number) => void;
+  onClose: () => void;
+}
+
+export const SamplingSiteMethodYupSchema = yup.object({
+  method_lookup_id: yup.number().required(),
+  description: yup.string().required(),
+  periods: yup
+    .array(
+      yup.object({
+        start_date: yup.string().isValidDateString().required(),
+        end_date: yup.string().isValidDateString().isEndDateSameOrAfterStartDate('start_date').required()
```

Comment:
Shouldn't we provide a string that user sees, letting them know how to fix this validation error?

Reply:
Both the isValidDateString and isEndDateSameOrAfterStartDate have default messages that get set if they fail. They can be overridden as well if need be

Reply:
Though, the required check should probably include a message "Start date is required" so it doesn't just default to "Required"

Reply:
Add some more text